src/register_hall_of_fame.c

```diff
@@ -0,0 +1,3067 @@
+        .vram = NNS_G2D_VRAM_TYPE_2DMAIN,
+        .resIdList = {
+            [GF_GFX_RES_TYPE_CHAR] = 55519,
+            [GF_GFX_RES_TYPE_PLTT] = 55520,
+            [GF_GFX_RES_TYPE_CELL] = 55512,
+            [GF_GFX_RES_TYPE_ANIM] = 55512,
+            [GF_GFX_RES_TYPE_MCEL] = -1,
+            [GF_GFX_RES_TYPE_MANM] = -1
+        },
+        .bgPriority = 2,
+        .vramTransfer =  0
+    },
+    [19] = {
+        .x = 125,
+        .y = 120,
+        .z =  0,
+        .animation =  0,
+        .spritePriority =  40,
+        .pal = 0,
+        .vram = NNS_G2D_VRAM_TYPE_2DMAIN,
+        .resIdList = {
+            [GF_GFX_RES_TYPE_CHAR] = 55519,
+            [GF_GFX_RES_TYPE_PLTT] = 55521,
+            [GF_GFX_RES_TYPE_CELL] = 55512,
+            [GF_GFX_RES_TYPE_ANIM] = 55512,
+            [GF_GFX_RES_TYPE_MCEL] = -1,
+            [GF_GFX_RES_TYPE_MANM] = -1
+        },
+        .bgPriority = 3,
+        .vramTransfer =  0
+    },
+    [20] = {
+        .x = 0,
+        .y = 0,
+        .z =  0,
+        .animation =  0,
+        .spritePriority =  0,
+        .pal = 0,
+        .vram = NNS_G2D_VRAM_TYPE_2DMAIN,
+        .resIdList = {
+            [GF_GFX_RES_TYPE_CHAR] = 55520,
+            [GF_GFX_RES_TYPE_PLTT] = 55522,
+            [GF_GFX_RES_TYPE_CELL] = 55515,
+            [GF_GFX_RES_TYPE_ANIM] = 55515,
+            [GF_GFX_RES_TYPE_MCEL] = -1,
+            [GF_GFX_RES_TYPE_MANM] = -1
+        },
+        .bgPriority = 2,
+        .vramTransfer =  0
+    }
+ };
+
+BOOL RegisterHallOfFame_Init(OVY_MANAGER *man, int *state) {
+    Main_SetVBlankIntrCB(NULL, NULL);
+    HBlankInterruptDisable();
+    GfGfx_EngineASetPlanes(0);
+    GfGfx_EngineBSetPlanes(0);
+    G2_BlendNone();
+    G2S_BlendNone();
+    GX_SetDispSelect(GX_DISP_SELECT_MAIN_SUB);
+    CreateHeap(HEAP_ID_3, HEAP_ID_REGISTER_HALL_OF_FAME, 0x50000);
+    RegisterHallOfFameData *data = (RegisterHallOfFameData *)OverlayManager_CreateAndGetData(man, sizeof(RegisterHallOfFameData), HEAP_ID_REGISTER_HALL_OF_FAME);
+    MI_CpuFill8(data, 0, sizeof(RegisterHallOfFameData));
+    data->args = OverlayManager_GetArgs(man);
+    data->msgData = NewMsgDataFromNarc(MSGDATA_LOAD_DIRECT, NARC_msgdata_msg, NARC_msg_msg_0180_bin, HEAP_ID_REGISTER_HALL_OF_FAME);
+    data->msgFormat = MessageFormat_New(HEAP_ID_REGISTER_HALL_OF_FAME);
+    data->strbuf1 = String_New(500, HEAP_ID_REGISTER_HALL_OF_FAME);
+    data->strbuf2 = String_New(500, HEAP_ID_REGISTER_HALL_OF_FAME);
+    data->narcA094 = NARC_New(NARC_a_1_0_1, HEAP_ID_REGISTER_HALL_OF_FAME);
+    data->narcA180 = NARC_New(NARC_a_1_8_0, HEAP_ID_REGISTER_HALL_OF_FAME);
+    RegisterHallOfFame_GetPartyDetails(data);
+    RegisterHallOfFame_SetGfxBanks();
+    RegisterHallOfFame_CreateBgConfig(data);
+    RegisterHallOfFame_CreateSpriteGfxHandlers(data);
+    sub_02004EC4(8, SEQ_GS_E_DENDOUIRI, 1);
+    sub_02004EC4(71, 0, 0);
+    LoadFontPal0(GF_PAL_LOCATION_MAIN_BG, (enum GFPalSlotOffset)0x1E0, HEAP_ID_REGISTER_HALL_OF_FAME);
```

Comment:
GF_PAL_SLOT_15_OFFSET, I know the pal slot offset enum is a bit stupid, but atm I would request it is used until removed